### PR TITLE
[cmake] add ENABLE_X11 option

### PR DESCRIPTION
**Pull request by akharche**
Implemented ENABLE_X11 to switch this flag through cmake

### CODE REVIEW

**Comment by akharche** [comment]
Option ENABLE_X11 should be set off on the environment without x11
Provided changes in product configs:
https://github.com/Intel-Media-SDK/product-configs/pull/80/files/208f425e90ae08c3497f89bd5e9ea8454a34accf

File: builder/FindPackages.cmake [hunk 1]
```diff
@@ -494,7 +494,6 @@ if( Linux )
   pkg_check_modules( PKG_MFX       libmfx>=1.28 )
 
   if ( PKG_X11_FOUND AND PKG_LIBVA_X11_FOUND )
```
Comment (dvrogozh) [inline builder/FindPackages.cmake]: If user explicitly sets ENABLE_X11=OFF then why we check X11 related packages? Please, modify the code as:
```
if(ENABLE_X11)
  pkg_check_modules( PKG_X11       x11 )
  pkg_check_modules( PKG_LIBVA_X11 libva-x11>=0.33 )
endif()

```

File: CMakeLists.txt [hunk 2]
```diff
@@ -58,6 +58,7 @@ option( ENABLE_OPENCL "Build targets dependent on OpenCL?" ON )
 option( ENABLE_ALL "Enable all dependencies and features?" OFF )
 
 if( CMAKE_SYSTEM_NAME MATCHES Linux )
+  option( ENABLE_X11 "Build X11 versions of the targets?" ON)
```
Comment (dvrogozh) [inline CMakeLists.txt]: I think that ENABLE_X11_DRI3 should depend from ENABLE_X11, use cmake_dependent_option.

File: builder/FindPackages.cmake [hunk 3]
```diff
@@ -494,7 +494,6 @@ if( Linux )
...
-    set( ENABLE_X11 true )
```
Comment (dvrogozh) [inline builder/FindPackages.cmake]: Design from the perspective: if ENABLE_X11=ON and X11 packages were not found - error out.

**Comment by dvrogozh** [comment]
This got outdated after recent https://github.com/Intel-Media-SDK/MediaSDK/pull/1385 merge. @akharche : I think I missed readme update in 1385 - can you, please, do?

**Pull request closed without merging**